### ISSUE TITLE
Support OmegaConf configs 

### ISSUE THREAD

**Issue by varun19299** [wandb/client#1233]
`wandb --version && python --version && uname`

* Weights and Biases version: 0.9.6 
* Python version: 3.8
* Operating System: MacOS Catalina

### Description

Can [OmegaConf](https://omegaconf.readthedocs.io/en/latest/#) objects be supported in addition to dictionaries?

Presently, OmegaConf configs donot get unrolled (or unpacked?) properly.

<img width="726" alt="image" src="https://user-images.githubusercontent.com/21169383/92875602-d2c0b280-f426-11ea-8661-c3262c1d314d.png">

Would expect: mlp.depth, mlp.hidden_size etc. in dot notation.

**Comment by cvphelps** [comment]
Hi there, this is an interesting feature request thank you for filing

**Comment by timoklein** [comment]
I was running into the same issue using [hydra](https://https://hydra.cc/). Would be cool if you guys could support this!

**Comment by varun19299** [comment]
Here's a workaround I'm presently using: converting the `DictConfig` explicitly to `dict`.

```
def cfg2dict(cfg: DictConfig) -> Dict:
    """
    Recursively convert OmegaConf to vanilla dict
    :param cfg:
    :return:
    """
    cfg_dict = {}
    for k, v in cfg.items():
        if type(v) == DictConfig:
            cfg_dict[k] = cfg2dict(v)
        else:
            cfg_dict[k] = v
    return cfg_dict

```
That said, it would be better to integrate with the attributes `wandb` explicitly accesses: since `DictConfigs` have [more or less all methods](https://github.com/omry/omegaconf/blob/master/omegaconf/dictconfig.py) of the `dict` class.

**Comment by varun19299** [comment]
Since we're also talking of hydra, here's another issue (I should probably open a separate issue thread for this):

Supporting hyperparam sweeps via hydra.

@timoklein: Do you use wandb or hydra's hyper parameter sweeping, and if so, what's your workflow like?

**Comment by omry** [comment]
If you want to convert OmegaConf containers to primitive containers, use `OmegaConf.to_container(cfg, resolve=True)`.
See the [docs of OmegaConf](https://omegaconf.readthedocs.io/en/latest/usage.html#omegaconf-to-container).

There is no need to create your own (broken, sorry) implementation :).

**Comment by varun19299** [comment]
Ah that's much better.

Yes, mine was just a hackish fix :).

**Comment by omry** [comment]
I agree with you that the best solution is to just accept DictConfig as a valid mapping.
A good way to do that is to use MutableMapping in the isisntance check:
```python
from typing import MutableMapping
from omegaconf import OmegaConf
d = {"foo": "bar"}
cfg = OmegaConf.create(d)
isinstance(d, MutableMapping)
>>> True
isinstance(cfg, MutableMapping)
>>> True
```

**Comment by timoklein** [comment]
> Since we're also talking of hydra, here's another issue (I should probably open a separate issue thread for this):
> 
> Supporting hyperparam sweeps via hydra.
> 
> @timoklein: Do you use wandb or hydra's hyper parameter sweeping, and if so, what's your workflow like?

Sorry, haven't used either of them. In my limited experience they don't work well together (yet).

**Comment by tomsercu** [comment]
+1 for supporting hydra / OmegaConf configs! See also #1052 
@varun19299 did you set something up that's working for you? I'm implementing now with hydra controlling the command line and hyperparam sweeps, and using wandb purely for logging, tracking, visualizing. Would love to hear your experience / MWEs

**Comment by varun19299** [comment]
I'm presently converting it to a dictionary, as suggested by @omry above (https://github.com/wandb/client/issues/1233#issuecomment-693205205).

I too use a similar setup: hydra takes care of all the config management, logging, working directories, hyperparam sweeps and multi-runs; with W&B for logging and visualisation. 

The slight annoyance is that hyper-param sweeps can easily swamp the project page, and it would be great if this could be integrated with W&B's sweep section (ie., if W&B supports external sweeping tools like Axsweeper, etc). @cvphelps is this something the W&B team would consider?

**Comment by ethanluoyc** [comment]
> Since we're also talking of hydra, here's another issue (I should probably open a separate issue thread for this):
> 
> Supporting hyperparam sweeps via hydra.
> 
> @timoklein: Do you use wandb or hydra's hyper parameter sweeping, and if so, what's your workflow like?

+1 for this.

**Comment by Trezorro** [comment]
I second this. Based on the docs, this kind of behaviour is almost expected to work.
Even with the workaround by @omry https://github.com/wandb/wandb/issues/1233#issuecomment-693205205 
the wandb config doesn't fully bahave as expected, as the dot style attribute access is broken. For some reason, all nested dictionaries will still be dictionaries when loaded from `wandb.config`, thus `wandb.config.foo.bar` will throw an attribute error.

**Comment by johndpope** [comment]
use this - this belongs on readme homepage

`
wandb.init(project='BLA', config=OmegaConf.to_container(config, resolve=True))
`
